### ISSUE TITLE
v1.0.0: Named return can be redeclared in an inner scope

### ISSUE THREAD

**Issue by horkhe** [go-lang-plugin-org/go-lang-idea-plugin#1151]
```Go
func bar() error {
	return nil
}

func Test() (err error) { // `err` declared here
	err := bar() // Error: `no new variable on left side of :=`, that is an error indeed.
	if err := bar(); err != nil { // Error: `no new variable on left side of :=`, That is not an error!
		return err
	} else {
		err := bar() // Error: `no new variable on left side of :=`, That is not an error!
		return err
	}
	return err
}
```
Same goes for any scope defining construct (e.g. for, switch, select)

**Comment by ignatov** [comment]
Partially fixed. A reminder for me: http://golang.org/ref/spec#Blocks

**Comment by ignatov** [comment]
Fixed in 45a99d1
